src/ACEtk/block/details/BaseDistributionData.hpp

```diff
@@ -0,0 +1,193 @@
+namespace details {
+
+/**
+ *  @class
+ *  @brief Base class for distribution data as a function of incident energy
+ */
+template< typename Distribution >
+class BaseDistributionData : protected details::Base {
+
+  /* fields */
+  std::size_t locb_;
+  InterpolationData interpolation_;
+  details::ColumnData data_;
+
+  /* auxiliary functions */
+  #include "ACEtk/block/details/BaseDistributionData/src/generateXSS.hpp"
+  #include "ACEtk/block/details/BaseDistributionData/src/verifyIncidentEnergyIndex.hpp"
+  #include "ACEtk/block/details/BaseDistributionData/src/generateBlocks.hpp"
+  #include "ACEtk/block/details/BaseDistributionData/src/verifySize.hpp"
+
+public:
+
+  /* constructor */
+  #include "ACEtk/block/details/BaseDistributionData/src/ctor.hpp"
+
+  /**
+   *  @brief Return the interpolation data
+   */
+  auto interpolationData() const { return this->interpolation_; }
+
+  /**
+   *  @brief Return the number of interpolation regions
+   */
+  std::size_t NB() const { return this->interpolationData().NB(); }
+
+  /**
+   *  @brief Return the number of interpolation regions
+   */
+  std::size_t numberInterpolationRegions() const { return this->NB(); }
+
+  /**
+   *  @brief Return the interpolation boundaries
+   */
+  auto NBT() const { return this->interpolationData().NBT(); }
+
+  /**
+   *  @brief Return the interpolation boundaries
+   */
+  auto boundaries() const { return this->NBT(); }
+
+  /**
+   *  @brief Return the interpolants
+   */
+  auto INT() const { return this->interpolationData().INT(); }
+
+  /**
+   *  @brief Return the interpolants
+   */
+  auto interpolants() const { return this->INT(); }
+
+  /**
+   *  @brief Return the number of incident energy values
+   */
+  std::size_t NE() const { return this->data_.N(); }
+
+  /**
+   *  @brief Return the number of incident energy values
+   */
+  std::size_t numberIncidentEnergies() const { return this->NE(); }
+
+  /**
+   *  @brief Return the incident energy values
+   */
+  auto incidentEnergies() const { return this->data_.column( 1 ); }
+
+  /**
+   *  @brief Return the incident energy for a given index
+   *
+   *  When the index is out of range an std::out_of_range exception is thrown
+   *  (debug mode only).
+   *
+   *  @param[in] index     the index (one-based)
+   */
+  double incidentEnergy( std::size_t index ) const {
+
+    #ifndef NDEBUG
+    this->verifyIncidentEnergyIndex( index );
+    #endif
+    return this->data_.value( 1, index );
+  }
+
+  /**
+   *  @brief Return the angular distribution locator for an incident
+   *         energy index
+   *
+   *  This locator is the value as stored in the XSS array.
+   *
+   *  When the index is out of range an std::out_of_range exception is thrown
+   *  (debug mode only).
+   *
+   *  @param[in] index     the index (one-based)
+   */
+  int LOCC( std::size_t index ) const {
+
+    #ifndef NDEBUG
+    this->verifyIncidentEnergyIndex( index );
+    #endif
+    return this->data_.value( 2, index );
```

Comment:
[low priority, style]
Return type of `this->data_.value` is `double`. Suggest `static_cast<int>`?

Reply:
Yes, but casting is not the right thing to do. Rounding is better.

This was changed in a later PR (I think it is in the fix/rounding one). It now says the following:
```
int LOCC( std::size_t index ) const {

    return round( this->value( this->values_.NC(), index ) );
  }
```

Reply:
round does return a double, so static_cast